docs/rules/no-invalid-this.md

```diff
@@ -0,0 +1,294 @@
+# Disallow `this` keyword on the outside of constructors/methods (no-invalid-this)
+
+To access `this` keyword on the outside of constructors/methods is dangerous.
+Under strict mode, `this` is `undefined`, so raises an `ReferenceError`.
```

Comment:
No, in strict mode, it's an undefined value. At the top level in a module, it's a reference error.